front/pages/w/[wId]/u/extract/_shared_schema_form.tsx

```diff
@@ -0,0 +1,418 @@
+import { PlusIcon } from "@heroicons/react/24/outline";
+import { useRouter } from "next/router";
+import React, { useState } from "react";
+
+import AppLayout from "@app/components/AppLayout";
+import { Button } from "@app/components/Button";
+import MainTab from "@app/components/use/MainTab";
+import { APIError } from "@app/lib/error";
+import { useEventSchemas } from "@app/lib/swr";
+import { classNames } from "@app/lib/utils";
+import {
+  eventSchemaPropertyTypeOptions,
+  EventSchemaType,
+} from "@app/types/extract";
+import { UserType, WorkspaceType } from "@app/types/user";
+
+type EventSchemaPropertyType = {
+  name: string;
+  type: string;
+  description: string;
+};
+
+export function ExtractEventSchemaForm({
+  user,
+  owner,
+  schema,
+  gaTrackingId,
+}: {
+  user: UserType | null;
+  owner: WorkspaceType;
+  schema?: EventSchemaType;
+  gaTrackingId: string;
+}) {
+  const { schemas } = useEventSchemas(owner);
+
+  const [marker, setMarker] = useState<string>(schema?.marker || "");
+  const [description, setDescription] = useState<string>(
+    schema?.description || ""
+  );
+  const [properties, setProperties] = useState<EventSchemaPropertyType[]>(
+    schema ? schema.properties : []
+  );
+
+  const [errorMarker, setErrorMarker] = useState<string>("");
+  const [errorDescription, setErrorDescription] = useState<string>("");
+  const [errorProperties, setErrorProperties] = useState<string>("");
+
+  const [isProcessing, setIsProcessing] = useState<boolean>(false);
+
+  const router = useRouter();
+
+  const handleFormValidation = (): boolean => {
+    let isFormValid = true;
+    if (!marker) {
+      isFormValid = false;
+      setErrorMarker("Marker is mandatory.");
+    }
+    if (!description) {
+      isFormValid = false;
+      setErrorDescription("Description is mandatory.");
+    }
+    if (properties.length === 0) {
+      isFormValid = false;
+      setErrorProperties("A template must contain at least one property.");
+    }
+    const hasEmptyProperties = properties.some(
+      (property: EventSchemaPropertyType) =>
+        !property.name || !property.type || !property.description
+    );
+    if (hasEmptyProperties) {
+      isFormValid = false;
+      setErrorProperties("All 3 fields are mandatory for each property added.");
+    }
+    if (!schema) {
```

Comment:
There's a unicity constraint on the db (unique marker per workspace) so this check is there only to gracefully display the error without raising an API error when we try to create it.